include/seqan3/alignment/pairwise/edit_distance_algorithm.hpp

```diff
@@ -13,10 +13,31 @@
 #pragma once
 
 #include <seqan3/alignment/pairwise/edit_distance_unbanded.hpp>
+#include <seqan3/core/concept/tuple.hpp>
 
 namespace seqan3::detail
 {
 
+/*!\brief A helper concept to check the input of the range based alignment algorithm interface.
+ * \tparam t The type to check.
+ *
+ * \details
+ *
+ * The following requirements are expected on a type that models indexed sequence pairs:
+ * * the type models std::ranges::forward_range,
+ * * the value type of this range models seqan3::tuple_like with exactly 2 elements,
+ * * the first type of the value type models seqan3::tuple_like with exactly 2 elements.
+ */
+template <typename t>
+SEQAN3_CONCEPT indexed_sequence_pair_range = requires (std::ranges::range_value_t<t> value)
```

Comment:
I guess that we later move this concept to a different file, because it is needed at some other place to.

Reply:
Yes, that is the plan. But we might as well put it already in some designated file.